explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,315 @@
+prefetched from the Document. PrefetchedSignedExchangeCache keeps the signed
+exchanges which were passed from the referrer Document which triggered the
+navigation to the current Document.
+1. While processing
+"[prefetch](https://html.spec.whatwg.org/multipage/links.html#link-type-prefetch)"
+link (eg: `<link rel="prefetch"
+href="https://cdn.feed.example/cdn.publisher.example/lib.js.sxg" as="script">`):
+   - If succeeded prefetching the signed exchange (including merkle integrity
+   check of the body), stores the parsedExchange which is the result of
+   [parsing a signed exchange](https://wicg.github.io/webpackage/loading.html#ref-for-parsing-a-signed-exchange)
+   (= inner request URL and inner response) to the SignedExchangeCache.
+1. While
+[navigating across documents](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents),
+copy the parsedExchanges in the SignedExchangeCache of the source document to
+the PrefetchedSignedExchangeCache of the target document. This is intended to
+provide a way to pass the cached signed exchange across origins even if the UA
+is using origin isolated HTTPCache mechanism.
+1. While processing
+[preload](https://html.spec.whatwg.org/multipage/links.html#link-type-preload)
+link HTTP headers (eg: Link: <https://cdn.publisher.example/lib.js>;
+rel="preload"; as="script"):
+   - Check whether matching "allowed-alt-sxg" link HTTP header
+   (`Link: <https://cdn.publisher.example/lib.js>;rel="allowed-alt-sxg";header-integrity="sha256-.."`)
+   exists or not. (Note that if the allowed-alt-sxg link HTTP header has
+   variants and variant-key attributes, the UA must execute the algorithm
+   written in
+   [HTTP Representation Variants spec](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+   to find the matching header.)
+   - Check whether all preload links which have matching allowed-alt-sxg link
+   header have matching (url and header-integrity) parsedExchange in
+   PrefetchedSignedExchangeCache. If the check passes, set request's stashed
+   exchange to the parsedExchange so the resource will be loaded from the cached
+   signed exchange.
+   - After processing the link HTTP headers, clears the
+   PrefetchedSignedExchangeCache.
+
+# Detailed design discussion
+## Header integrity of signed exchange
+We use the SHA256 hash value of *signedHeaders* in
+[application/signed-exchange format](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#application-signed-exchange)
+for integrity checking. This signedHeaders is *"the canonical serialization of
+the CBOR representation of the response headers of the exchange represented by
+the application/signed-exchange resource, excluding the Signature header
+field"*. So this value doesn’t change even if the publisher signs the content
+again or changes the signing key. This header-integrity value can verify the
```

Comment:
```suggestion
again or changes the signing key. This header-integrity value also guarantees the content body hasn't changed, because the signed headers are required to include a `Digest` header with a hash of the content body.
```